src/util/strencodings.cpp

```diff
@@ -46,14 +46,20 @@ bool IsHex(std::string_view str)
     return (str.size() > 0) && (str.size()%2 == 0);
 }
 
-bool IsHexNumber(std::string_view str)
+std::optional<std::string> TrySanitizeHexNumber(std::string_view input, int result_size)
 {
-    if (str.substr(0, 2) == "0x") str.remove_prefix(2);
-    for (char c : str) {
-        if (HexDigit(c) < 0) return false;
+    input = util::RemovePrefixView(input, "0x");
+    if (result_size < 0) result_size = input.size(); // negative result_size disables the size check
+    if (input.empty() || ((int)input.size() > result_size)) return std::nullopt;
+    for (char c : input) {
+        if (HexDigit(c) < 0) return std::nullopt;
     }
```

Comment:
could we use `IsHex` here instead, i.e. merge it with previous line:
```C++
    if (input.empty() || ((int)input.size() > result_size) || !IsHex(input)) return std::nullopt;
```

though this contains an extra evennes requirement which we may not want here.